Source/WebCore/loader/PolicyChecker.cpp

```diff
@@ -298,10 +298,13 @@ void PolicyChecker::checkNavigationPolicy(ResourceRequest&& request, const Resou
 
     if (isInitialEmptyDocumentLoad) {
         // We ignore the response from the client for initial empty document loads and proceed with the load synchronously.
-        frameLoader->client().dispatchDecidePolicyForNavigationAction(action, request, redirectResponse, formState.get(), clientRedirectSourceForHistory, navigationID, hitTestResult(action), hasOpener, isPerformingHTTPFallback, sandboxFlags, policyDecisionMode, [](PolicyAction) { });
+        frameLoader->client().dispatchDecidePolicyForNavigationAction(action, request, redirectResponse, formState.get(), clientRedirectSourceForHistory, navigationID,
+            hitTestResult(action), hasOpener, isPerformingHTTPFallback, sandboxFlags, mayNeedBeforeUnloadPrompt, policyDecisionMode, [](PolicyAction) { });
         decisionHandler(PolicyAction::Use);
-    } else
-        frameLoader->client().dispatchDecidePolicyForNavigationAction(action, request, redirectResponse, formState.get(), clientRedirectSourceForHistory, navigationID, hitTestResult(action), hasOpener, isPerformingHTTPFallback, sandboxFlags, policyDecisionMode, WTFMove(decisionHandler));
+    } else {
+        frameLoader->client().dispatchDecidePolicyForNavigationAction(action, request, redirectResponse, formState.get(), clientRedirectSourceForHistory, navigationID,
+            hitTestResult(action), hasOpener, isPerformingHTTPFallback, sandboxFlags, mayNeedBeforeUnloadPrompt, policyDecisionMode, WTFMove(decisionHandler));
```

Comment:
nit: this is an odd place to put a newline.  I'd either keep them all on one line or put them each on their own line.  Same above.